src/distro/distro.py

```diff
@@ -738,8 +749,22 @@ def __init__(
                 self.os_release_file = usr_lib_os_release_file
 
         self.distro_release_file = distro_release_file or ""  # updated later
-        self.include_lsb = include_lsb
-        self.include_uname = include_uname
+
+        is_root_dir_defined = root_dir is not None
+        if is_root_dir_defined and (include_lsb or include_uname or include_oslevel):
+            logging.getLogger(__name__).warning(
+                "including subprocess data sources from specific root_dir may not work"
+                " as expected."
+            )
```

Comment:
@HorlogeSkynet is there a chance for the subprocesses to return the right data (other than by coincidence) *ever*? If not, can we turn that into a raise of `ValueError`?

Reply:
Good point. In "real" environments (a.k.a not against functional tests resources), I don't think so either.
Any idea to replace the error message ?
Would `Including subprocess data sources from specific root_dir is disallowed to prevent misleading information` do the job ?

Reply:
@HorlogeSkynet I'd replace `misleading` by plain `false` personally but otherwise that seems great, thanks! :+1:

Reply:
@HorlogeSkynet thanks for your work and patience! :+1: